### PR TITLE
feat: full newsletter support

### PR DESCRIPTION
**Pull request by vinikjkkj**
## How newsletter works

The only difference between ordinary messages and these messages is that they are not encrypted

Each message has two IDs, the common id and the "server_id", which is the id used for reactions and views, to react an message, you need to use "server_id"

The default connection does not receive newsletter updates, you need to subscribe to receive reaction and views events, I have not tested this in other events, apparently messages don't need this

### How to subscribe

Use: `sock.subscribeNewsletterUpdates(jid)`

### Functions

- subscribeNewsletterUpdates
- newsletterReactionMode
- newsletterUpdateDescription
- newsletterUpdateName
- newsletterUpdatePicture
- newsletterRemovePicture
- newsletterUnfollow
- newsletterFollow
- newsletterUnmute
- newsletterMute
- newsletterCreate
- newsletterMetadata
- newsletterAdminCount
- newsletterChangeOwner
- newsletterDemote
- newsletterDelete
- newsletterReactMessage
- newsletterfetchMessages
- newsletterfetchUpdates

## Important

I want you guys to test to see if everything is ok and let me know if something doesn't work

### CODE REVIEW

File: src/Socket/messages-recv.ts [hunk 1]
```diff
@@ -325,6 +325,59 @@ export const makeMessagesRecvSocket = (config: SocketConfig) => {
...
+        }
+	}
+
+	const handleMexNewsletterNotification = (id: string, node: BinaryNode) => {
+		const operation = node?.attrs.op_name
+		const content = JSON.parse(node?.content?.toString()!)
+
+		let contentPath
+
+		if(operation === MexOperations.PROMOTE || operation === MexOperations.DEMOTE){
```
Comment (azudindaem) [inline src/Socket/messages-recv.ts:376]: need to simplify the logic

```ts
if(operation === MexOperations.PROMOTE){
	action = 'promote'
	contentPath = content.data[XWAPaths.PROMOTE]
	ev.emit('newsletter-participants.update', //add rest 
}

if(operation === MexOperations.DEMOTE){
	action = 'demote'
	contentPath = content.data[XWAPaths.DEMOTE]
	ev.emit('newsletter-participants.update', //add rest 
}

if(operation === MexOperations.UPDATE){
	contentPath = content.data[XWAPaths.METADATA_UPDATE]
	ev.emit('newsletter-settings.update', {id, update: contentPath.thread_metadata.settings as NewsletterSettingsUpdate})
}
```

**Review by azudindaem — CHANGES_REQUESTED** [review]

**Comment by azudindaem** [comment]
Wee need to do full test for this PR, will update later.. please ensure our code format and variable well formatted

**Comment by PurpShell** [comment]
@donaldjansen

**Comment by donaldjansen** [comment]
newsletter create causes the baileys to crash.
image

File: src/Socket/messages-recv.ts (hunk 1 above)
Comment (vinikjkkj) [reply src/Socket/messages-recv.ts:376]: i think is better to use switch case, what do you think?

Comment (purpshell) [reply src/Socket/messages-recv.ts:376]: yes use a switch

**Comment by manojsitapara** [comment]
@vinikjkkj How to get list of newsletter which provide us newsletter id ("123456789@newsletter") in which I am admin? similar to how we get list of groups

**Comment by manojsitapara** [comment]
I think current PR does not support post image, it send blank image

**Comment by w3nder** [comment]
> I think current PR does not support post image, it send blank image

same

**Comment by vihangaashin** [comment]
Screenshot_20240605-144611_WhatsApp
When I use bailyes to send an image to a whatsapp channel, this is how the preview is shown before the image is downloaded. How to fix this?

**Comment by donaldjansen** [comment]
> Screenshot_20240605-144611_WhatsApp
> When I use bailyes to send an image to a whatsapp channel, this is how the preview is shown before the image is downloaded. How to fix this?

This newsletter version does not yet support sending media to a newsletter/channel.
I'm sure it will be implemented soon. There are quite a few differences between sending messages and sending newsletter messages.

**Comment by shizothetechie** [comment]
I tested & it works fine ✅

**Comment by Riders004** [comment]
When this got merged

**Comment by Riders004** [comment]
> I tested & it works fine ✅

When it will get merged

**Comment by Riders004** [comment]
Using this only working with owner command but not with other admin command

**Comment by Riders004** [comment]
> Using this only working with owner command but not with other admin command
> 
> Screenshot_2024-07-05-21-04-58-222_com whatsapp
> 
> Screenshot_2024-07-05-21-04-55-479_com whatsapp

@vinikjkkj @PurpShell @PurpShell  @shizothetechie

**Comment by joweste** [comment]
How can I list all newsletters? I just created one but I don’t know how I get your metadata

**Comment by Sithuwili-dev** [comment]
How send messages

**Comment by Riders004** [comment]
> How send messages

Use m.chat

**Comment by shizothetechie** [comment]
> Using this only working from channel owner or bot hosting number command but not from other admin or bot owner command
> 
> Screenshot_2024-07-05-21-04-58-222_com whatsapp
> 
> Screenshot_2024-07-05-21-04-55-479_com whatsapp

Database error 
maybe You need to Seprate the database for newsletter then only it will work for all other admins 

try implementing Separate Database for newsletter

**Comment by Riders004** [comment]
> > Using this only working from channel owner or bot hosting number command but not from other admin or bot owner command
> > Screenshot_2024-07-05-21-04-58-222_com whatsapp
> > Screenshot_2024-07-05-21-04-55-479_com whatsapp
> 
> Database error maybe You need to Seprate the database for newsletter then only it will work for all other admins
> 
> try implementing Separate Database for newsletter

How

**Comment by shizothetechie** [comment]
> > > Using this only working from channel owner or bot hosting number command but not from other admin or bot owner command
> > > Screenshot_2024-07-05-21-04-58-222_com whatsapp
> > > Screenshot_2024-07-05-21-04-55-479_com whatsapp
> > 
> > 
> > Database error maybe You need to Seprate the database for newsletter then only it will work for all other admins
> > try implementing Separate Database for newsletter
> 
> How

Thats Depends upon your system 
do that according to your Normal database management

**Comment by Riders004** [comment]
> > > > Using this only working from channel owner or bot hosting number command but not from other admin or bot owner command
> > > > Screenshot_2024-07-05-21-04-58-222_com whatsapp
> > > > Screenshot_2024-07-05-21-04-55-479_com whatsapp
> > > 
> > > 
> > > Database error maybe You need to Seprate the database for newsletter then only it will work for all other admins
> > > try implementing Separate Database for newsletter
> > 
> > 
> > How
> 
> Thats Depends upon your system do that according to your Normal database management

How to do in Cheems bot 😃

**Comment by Riders004** [comment]
subscribeNewsletterUpdates

What the use of this ?

**Comment by Riders004** [comment]
newsletterFollow us working only with jid I'd not with channel link and newsletterMetadata have some error 
@vinikjkkj

**Pull request merged**